core/cat/mad_hatter/plugin.py

```diff
@@ -101,40 +101,37 @@ def deactivate(self):
         self._tools = []
         self._forms = []
         self._deactivate_endpoints()
-        self._plugin_overrides = []
+        self._plugin_overrides = {}
         self._active = False
 
     # get plugin settings JSON schema
     def settings_schema(self):
         # is "settings_schema" hook defined in the plugin?
-        for h in self._plugin_overrides:
-            if h.name == "settings_schema":
-                return h.function()
-            else:
-                # if the "settings_schema" is not defined but
-                # "settings_model" is it get the schema from the model
-                if h.name == "settings_model":
-                    return h.function().model_json_schema()
+        if "settings_schema" in self._plugin_overrides:
+            return self._plugin_overrides["settings_schema"]()
```

Comment:
please, consider that `self._plugin_overrides` becomes a `Dict[str, CatPluginDecorator]`. Therefore, each time you want to run the function related to an item of `self._plugin_overrides`, you should run `self._plugin_overrides[key].function()`